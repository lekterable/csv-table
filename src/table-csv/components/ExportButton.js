import React from 'react'
import { CSVLink } from 'react-csv'

export default ({ data = [] }) => {
  return (
    <CSVLink
      data-testid="export-button"
      tabIndex="-1"
      className="export-link"
      data={data}
      filename={'react-csv.csv'}
    >
      <input
        className="button button--secondary"
        type="button"
        value="Export"
      />
    </CSVLink>
  )
}
