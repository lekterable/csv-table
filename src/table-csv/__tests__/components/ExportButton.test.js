import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render } from 'react-testing-library'
import ExportButton from '../../components/ExportButton'

describe('<ExportButton />', () => {
  afterEach(cleanup)

  it('should match snapshot', () => {
    const component = renderer.create(<ExportButton />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should render export button', () => {
    const { queryByTestId } = render(<ExportButton />)
    const exportButton = queryByTestId('export-button')

    expect(exportButton).not.toBeNull()
  })
})
