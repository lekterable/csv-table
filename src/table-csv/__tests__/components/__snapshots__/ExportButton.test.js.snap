// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExportButton /> should match snapshot 1`] = `
<a
  className="export-link"
  data-testid="export-button"
  download="react-csv.csv"
  href="data:text/csv;charset=utf-8,ï»¿"
  onClick={[Function]}
  tabIndex="-1"
  target="_self"
>
  <input
    className="button button--secondary"
    type="button"
    value="Export"
  />
</a>
`;
